src/main/webapp/resources/js/utilities/array-utilities.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Splits an array into multiple, smaller arrays
+ *
+ * @param items the array to be chunked
+ *
+ * @returns a chunked array
+ */
+export function chunkArray<T>(items: T[]) {
+  const size = calculateChunkSize(items.length);
+  const clone = [...items];
+  const chunks = [];
+
+  while (clone.length) {
+    chunks.push(clone.splice(0, size));
+  }
+
+  return chunks;
+}
+
+/**
+ * Calculate the best request chunk size
+ *
+ * @param arraySize the size of the array
+ *
+ * @returns the chunk size
+ */
+function calculateChunkSize(arraySize: number) {
+  const MIN = 10;
+  const MAX = 100;
+  const estimated = Math.ceil(arraySize / 10);
+  return estimated < MIN ? MIN : estimated > MAX ? MAX : estimated;
```

Comment:
Can you make this line easier to read?  Linting is giving me the following error: `Extract this nested ternary operation into an independent statement.`

Reply:
Please see e17ee8dfff84ca10254ec8396a555f20da128916.

Reply:
Perfect, thanks